safe-room.lic

```diff
@@ -63,6 +66,31 @@ class SafeRoom
     score > @health_threshold
   end
 
+  def devour_healing(settings)
+    return unless settings.necromancer_healing['Devour']
+    return unless settings.thanatology['harvest_container']
+    
+    walk_to(settings.safe_room)
+    release_invisibility
+    devours = count_items_in_container('material', settings.thanatology['harvest_container'])
+    return unless devours > 0
+
+    stow_hands
+    return if right_hand
+    
+    while need_healing? && devours > 0
+      while mana < 40
+        echo('Waiting on mana...')
+        pause 10
+      end
+
+      result = bput("get material in my #{settings.thanatology['harvest_container']}", 'You get', 'You are already', 'What were you') unless right_hand
+      break if result =~ 'What were you'
+      DRCA.cast_spell(settings.necromancer_healing['Devour'], settings)
+      devours -= 1 unless right_hand
+    end
```

Comment:
Do we want to put material back into the harvest container in case we finish healing with some left in hand?

Reply:
Good call, added it. Though it shouldn't pull out a second if there is already some in hand. I use the `unless right_hand` to avoid getting another out.